### PR TITLE
Add support for sauce labs

### PR DESCRIPTION
**Pull request by webteckie**
Add support for sauce labs as the prefer method to do nightwatch ui/functional testing in the Travis CI environment.  This initial attempt may or may not work as there's quite a bit of setup that had to happen to make this work.

### CODE REVIEW

**Comment by webteckie** [comment]
@alsoicode any ideas why the pagination unit test is failing with:
```
  1) When paginating results with an optional expression should return results plus query metadata and pagination metadata:
     MongoError: exception: invalid operator: $search
      at Function.MongoError.create (node_modules/mongoose/node_modules/mongodb/node_modules/mongodb-core/lib/error.js:31:11)
      at commandCallback (node_modules/mongoose/node_modules/mongodb/node_modules/mongodb-core/lib/topologies/server.js:929:66)
      at Callbacks.emit (node_modules/mongoose/node_modules/mongodb/node_modules/mongodb-core/lib/topologies/server.js:116:3)
      at null.messageHandler (node_modules/mongoose/node_modules/mongodb/node_modules/mongodb-core/lib/topologies/server.js:282:23)
      at Socket.<anonymous> (node_modules/mongoose/node_modules/mongodb/node_modules/mongodb-core/lib/connection/connection.js:273:22)
      at readableAddChunk (_stream_readable.js:146:16)
      at Socket.Readable.push (_stream_readable.js:110:10)
      at TCP.onread (net.js:523:20)
```

**Comment by morenoh149** [comment]
just a nitpick. Try to make PR's be 1 commit (unless each commit really addresses different concerns but then they should probably be multiple PR's). See [squash](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html).

**Comment by webteckie** [comment]
@morenoh149 i know that's a goal.  However, when you are dealing with a remote environment that behaves different than the local it takes some trial and error to get things right.

**Comment by alsoicode** [comment]
@webteckie Hmm. Perhaps it's the version of MongoDB? If memory serves, my local version was 3.0.6 when I added the optional expression for pagination.

**Comment by morenoh149** [comment]
I just tested this branch and `npm run test-cov` passes fine. I have `mongod --version
db version v3.2.1`

**Comment by webteckie** [comment]
Yeah, I use 3.2 locally and it works fine as well.   Anybody knows what version travis uses?  If not,  I think I'm going to edit the PR to dump the mongo version...more trial and error ☺

**Comment by morenoh149** [comment]
I find this odd as it seems there is an attempt to load in mongo v3 here https://github.com/keystonejs/keystone/blob/master/.travis.yml#L32

**Comment by webteckie** [comment]
I think the extra adons tag I added probably hided the one for mongo.  I've consolidated the addons.  However, now I'm getting the followin...which I didn't change that file...travis hates me :-)
```
/home/travis/build/keystonejs/keystone/fields/mixins/ArrayField.js
  122:3  error  Missing trailing comma  comma-dangle
```

**Comment by morenoh149** [comment]
maybe one of your new tests are exercising the linting rules. [That line](https://github.com/keystonejs/keystone/blob/master/fields/mixins/ArrayField.js#L122) does need to have a trailing comma as our [eslint config](https://github.com/keystonejs/keystone/blob/master/.eslintrc#L21) specifies. Keep in mind the CI tests have been broken for a while ;)

File: .travis.yml [hunk 1]
```diff
@@ -8,6 +8,7 @@ before_script:
 
 script:
   - npm run test-cov
+  - npm run test-e2e-saucelabs
 
 sudo: false
```
Comment (morenoh149) [inline .travis.yml]: try changing this line to `sudo: required`

**Comment by webteckie** [comment]
@morenoh149 ah crap I I didn't realize travis merge PR with master for its builds and so I was missing the commit fro @erg0dic to ArrayField.js. Will update my branch to latest and try again :-)

**Pull request closed without merging**